code/modules/projectiles/guns/syringe_gun.dm

```diff
@@ -95,3 +94,249 @@
+	icon_state = "rapidsyringegun"
+	max_syringes = 14  // full two boxes worth
+	/// Internal reagent container.
+	var/obj/item/reagent_containers/glass/beaker/bluespace/internal_beaker
+	/// Index of possible transfer amounts to select
+	var/reagents_per_syringe_selection = 1
+	/// Possible options for alt-clicking the
+	var/list/possible_transfer_amounts = list(5, 10, 15)
+
+/obj/item/gun/syringe/rapidsyringe/Initialize(mapload)
+	. = ..()
+	internal_beaker = new(src)
+
+/obj/item/gun/syringe/rapidsyringe/Destroy()
+	for(var/syringe in syringes)
+		qdel(syringe)
+
+	syringes = null
+
+	if(chambered)
+		if(chambered.BB)
+			qdel(chambered.BB)
+		qdel(chambered)
+
+	QDEL_NULL(internal_beaker)
+	. = ..()
+
+/obj/item/gun/syringe/rapidsyringe/proc/get_units_per_shot()
+	return possible_transfer_amounts[reagents_per_syringe_selection]
+
+/obj/item/gun/syringe/rapidsyringe/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>Use in hand to unload an empty syringe.</span>"
+	. += "<span class='notice'>Alt-click to toggle the amount of reagents per shot.</span>"
+	. += "A switch on the side is set to [get_units_per_shot()] unit\s per shot."
+	. += "Its internal reservoir contains [internal_beaker.reagents.total_volume]/[internal_beaker.reagents.maximum_volume] units."
+	if(chambered.BB)
+		. += "<span class='notice'>You can see the chambered syringe contains [chambered.BB.reagents.total_volume] units."
+
+/// If user is null in this proc, messages will just be ignored
+/obj/item/gun/syringe/rapidsyringe/proc/insert_single_syringe(obj/item/reagent_containers/syringe/new_syringe, mob/user)
+	if(new_syringe.reagents.total_volume)
+		if(user)
+			to_chat(user, "<span class='warning'>[src] only accepts empty syringes.</span>")
+		return FALSE
+	var/in_clip = length(syringes) + (chambered?.BB ? 1 : 0)
+	if(in_clip < max_syringes)
+		if(user)
+			if(!user.unEquip(new_syringe))
+				return
+			to_chat(user, "<span class='notice'>You load \the [new_syringe] into [src].</span>")
+		syringes.Add(new_syringe)
+		new_syringe.loc = src
+		process_chamber() // Chamber the syringe if none is already
+		return TRUE
+	else
+		if(user)
+			to_chat(user, "<span class='warning'>[src] is full!</span>")
+		return FALSE
+
+/obj/item/gun/syringe/rapidsyringe/attackby(obj/item/A, mob/user, params, show_msg)
+
+	if(istype(A, /obj/item/storage))
+		// Boxes can be dumped in.
+		var/obj/item/storage/container = A
+		if(!length(container.contents))
+			to_chat(user, "<span class='warning'>[A] is empty!</span>")
+			return TRUE
+		if(length(syringes) + (chambered?.BB ? 1 : 0) == max_syringes)
+			to_chat(user, "<span class='warning'>[src] is full!</span>")
+			return TRUE
+
+		var/total_inserted = 0
+		for(var/obj/item/reagent_containers/syringe/S in container)
+			if((length(syringes) + (chambered?.BB ? 1 : 0)) == max_syringes)
+				// full
+				break
+
+			// only allow empty syringes
+			if(!S.reagents.total_volume)
+				syringes += S
+				S.loc = src
```

Comment:
loc override here.

Reply:
What do you mean, just like make it forcemove?